tanjun/checks.py

```diff
@@ -480,6 +480,10 @@ async def __call__(
         if ctx.guild_id is None:
             permissions = utilities.DM_PERMISSIONS
 
+        elif isinstance(ctx, tanjun.SlashContext):
+            assert ctx.interaction.app_permissions is not None
```

Comment:
TODO: check that this is infact provided when the app is in a guild without a bot user

Reply:
it is